packages/prisma/extensions/audit-log-custom-method-client-extension.ts

```diff
@@ -0,0 +1,183 @@
+import { Prisma } from "@prisma/client";
+
+import saveAuditLogWorker from "@calcom/features/audit-log/saveAuditLogWorker";
+import { BookingAuditLogOption } from "@calcom/features/audit-log/types/BookingAuditLogTypes";
+import { EventTypeAuditLogOption } from "@calcom/features/audit-log/types/EventTypeAuditLogTypes";
+import { prisma } from "@calcom/prisma";
+
+export function auditLogCustomMethodClientExtension() {
+  return Prisma.defineExtension({
+    query: {
+      user: {
+        async delete({ args, query }) {
+          const deletedUser = await query(args); //
+          const auditLogFeature = await prisma.feature.findUnique({ where: { slug: "audit-log" } });
```

Comment:
We should have a migration that adds the record as false by default so admins can toggle it without querying the DB directly. Here's an example: https://github.com/calcom/cal.com/blob/main/packages/prisma/migrations/20230329000000_add_insights_feature_flag/migration.sql

Reply:
As you have read on the above the feature flag it converted into process.env.AUDITLOGENABLE

Reply:
Done.

Reply:
Done. But using .env flag